lib/policyEvaluator/utils/conditions.js

```diff
@@ -0,0 +1,407 @@
+                const numberItem = Number.parseInt(item, 10);
+                if (Number.isNaN(numberItem)) {
+                    return false;
+                }
+                return numberKey >= numberItem;
+            });
+        },
+        DateEquals: function dateEquals(key, value) {
+            const epochKey = convertToEpochTime(key);
+            const epochValues = convertToEpochTime(value);
+            // eslint-disable-next-line new-cap
+            return operatorMap.NumericEquals(epochKey, epochValues);
+        },
+        DateNotEquals: function dateNotEquals(key, value) {
+            const epochKey = convertToEpochTime(key);
+            const epochValues = convertToEpochTime(value);
+            // eslint-disable-next-line new-cap
+            return operatorMap.NumericNotEquals(epochKey, epochValues);
+        },
+        DateLessThan: function dateLessThan(key, value) {
+            const epochKey = convertToEpochTime(key);
+            const epochValues = convertToEpochTime(value);
+            // eslint-disable-next-line new-cap
+            return operatorMap.NumericLessThan(epochKey, epochValues);
+        },
+        DateLessThanEquals: function dateLessThanEquals(key, value) {
+            const epochKey = convertToEpochTime(key);
+            const epochValues = convertToEpochTime(value);
+            // eslint-disable-next-line new-cap
+            return operatorMap.NumericLessThanEquals(epochKey, epochValues);
+        },
+        DateGreaterThan: function dateGreaterThan(key, value) {
+            const epochKey = convertToEpochTime(key);
+            const epochValues = convertToEpochTime(value);
+            // eslint-disable-next-line new-cap
+            return operatorMap.NumericGreaterThan(epochKey, epochValues);
+        },
+        DateGreaterThanEquals: function dateGreaterThanEquals(key, value) {
+            const epochKey = convertToEpochTime(key);
+            const epochValues = convertToEpochTime(value);
+            // eslint-disable-next-line new-cap
+            return operatorMap.NumericGreaterThanEquals(epochKey, epochValues);
+        },
+        Bool: function bool(key, value) {
+            // Tested with policy validator and it just appears to be a string
+            // comparison (can send in values other than true or false)
+            // eslint-disable-next-line new-cap
+            return operatorMap.StringEquals(key, value);
+        },
+        BinaryEquals: function binaryEquals(key, value) {
+            const base64Key = new Buffer(key).toString('base64');
+            return value.some(item => item === base64Key);
+        },
+        BinaryNotEquals: function binaryNotEquals(key, value) {
+            // eslint-disable-next-line new-cap
+            return !operatorMap.BinaryEquals(key, value);
+        },
+        IpAddress: function ipAddress(key, value) {
+            return value.some(item => {
+                let inRange = false;
+                try {
```

Comment:
For V8 performance reasons, can you extract the try...catch into an isolated function?